utils/ort/src/test/kotlin/UtilsTest.kt

```diff
@@ -379,4 +387,51 @@ class UtilsTest : WordSpec({
             normalizeVcsUrl(url) shouldBe url
         }
     }
+
+    "requestPasswordAuthentication" should {
+        "return a correct authentication object for a host, port, and scheme combination" {
+            val host = "www.example.org"
+            val port = 442
+            val scheme = "https"
+
+            mockkStatic("org.ossreviewtoolkit.utils.ort.UtilsKt")
+            mockkStatic(Authenticator::class)
+            val proxySelector = mockk<OrtProxySelector>()
+            val passwordAuth = mockk<PasswordAuthentication>()
+
+            every { installAuthenticatorAndProxySelector() } returns proxySelector
+            every {
+                Authenticator.requestPasswordAuthentication(host, null, port, scheme, null, null)
+            } returns passwordAuth
```

Comment:
Hmm. I'm having difficulties appreciating the value of these tests. You're basically mocking everything that's being called in `requestPasswordAuthentication()` away, effectively reimplementing the helper function with mocks only, leaving no original code to test.

Wouldn't it make more sense to simply "set" environment variables via `Os.env` and reply on `EnvVarAuthenticator` to do its job?

Reply:
> not more and not less.

I believe the test *does* more, in the sense that it mocks unnecessary things. For example, I believe lines 399 and 402 could simply be dropped without sacrificing anything.